qiita_db/util.py

```diff
@@ -255,53 +255,53 @@ def get_emp_status(key='emp_status'):
 
 
 def create_rand_string(length, punct=True):
-        """Returns a string of random ascii characters
+    """Returns a string of random ascii characters
 
-        Parameters
-        ----------
-        length: int
-            Length of string to return
-        punct: bool, optional
-            Include punctiation as well as letters and numbers. Default True.
-        """
-        chars = ''.join((ascii_letters, digits))
-        if punct:
-            chars = ''.join((chars, punctuation))
-        return ''.join(choice(chars) for i in range(length))
+    Parameters
+    ----------
+    length: int
+        Length of string to return
+    punct: bool, optional
+        Include punctiation as well as letters and numbers. Default True.
+    """
+    chars = ''.join((ascii_letters, digits))
+    if punct:
+        chars = ''.join((chars, punctuation))
+    return ''.join(choice(chars) for i in range(length))
 
 
 def hash_password(password, hashedpw=None):
-        """ Hashes password
-
-        Parameters
-        ----------
-        password: str
-            Plaintext password
-        hashedpw: str, optional
-            Previously hashed password for bcrypt to pull salt from. If not
-            given, salt generated before hash
-
-        Returns
-        -------
-        str
-            Hashed password
-
-        Notes
-        -----
-        Relies on bcrypt library to hash passwords, which stores the salt as
-        part of the hashed password. Don't need to actually store the salt
-        because of this.
-        """
-        # all the encode/decode as a python 3 workaround for bcrypt
-        if hashedpw is None:
-            hashedpw = gensalt()
-        else:
-            hashedpw = hashedpw.encode('utf-8')
-        password = password.encode('utf-8')
-        output = hashpw(password, hashedpw)
-        if isinstance(output, bytes):
-            output = output.decode("utf-8")
-        return output
+    """ Hashes password
+
+    Parameters
+    ----------
+    password: str
+        Plaintext password
+    hashedpw: str, optional
+        Previously hashed password for bcrypt to pull salt from. If not
+        given, salt generated before hash
+
+    Returns
+    -------
+    str
+        Hashed password
+
+    Notes
+    -----
+    Relies on bcrypt library to hash passwords, which stores the salt as
+    part of the hashed password. Don't need to actually store the salt
+    because of this.
+    """
+    # all the encode/decode as a python 3 workaround for bcrypt
+    if hashedpw is None:
+        hashedpw = gensalt()
+    else:
+        hashedpw = hashedpw.encode('utf-8')
```

Comment:
Ah, this is also because we're using the bcrypt library directly. the `hashpw` needs the string and a salt. This salt can either be a new salt if it's not comparing anything, or the hashed password and it pulls the hash from that password for comparison later.